### PR TITLE
feat: Run `pre-commit` framework for all Git repositories

### PR DESCRIPTION
**Pull request by MoritzWeber0**
This adds the "local hooks" feature, described in https://github.com/DSD-DBS/capella-dockerimages/issues/211.
To use it, add a `.pre-commit-config.yaml` to a Git repository.

Git will then automatically run the defined hooks.

### CODE REVIEW

File: docs/docs/git-hooks/git-hooks.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+The implementation was tested with the official Git CLI and JGit/EGit (in Eclipse).
+
+## Project-Specific Hooks
+
+Project-specific hooks can be defined separately for each Git repository. The defined Git hooks will not be auto-updated by us. Instead, the responsibility falls on the repository owners to keep them up-to-date and maintain them.
+
+### Installation
+
+To install Git hooks, add a `.pre-commit-config.yaml` file to the root of your repository. Simply follow the official pre-commit instructions: [Add a pre-commit configuration](https://pre-commit.com/#2-add-a-pre-commit-configuration). Commit the file after making changes. Git will automatically detect it, eliminating the need to install the Git hooks script in the Docker images.
```
Comment (amolenaar) [inline docs/docs/git-hooks/git-hooks.md:18]: The last sentence is a bit weird. I suppose you mean that pre-commit will automatically be triggered if the file exists and no explicit `pre-commit install` command is required.

File: docs/docs/git-hooks/git-hooks.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      hooks:
+        - id: my-hook
+          stages: [pre-commit]
+    ```
+
+    For more information, refer to the `pre-commit` documentation: [Confining hooks to run at certain stages](https://pre-commit.com/#confining-hooks-to-run-at-certain-stages).
+
+### Updates
+
+To auto-update Git hooks, update the versions in your repository's `.pre-commit-config.yaml` file. Refer to the official documentation for [updating hooks automatically](https://pre-commit.com/#updating-hooks-automatically).
```
Comment (amolenaar) [inline docs/docs/git-hooks/git-hooks.md:42]: When in a Capella session, you do not have access to a command line, right? Then there's no way to update. updates could be done through a Jupyter session.

File: base/Dockerfile [hunk 3]
```diff
@@ -42,19 +42,45 @@ RUN useradd -l -m -u $UID techuser && \
...
+        -t commit-msg \
+        -t post-checkout \
+        -t post-commit \
+        -t post-merge \
+        -t post-rewrite \
+        -t pre-commit \
+        -t pre-merge-commit \
+        -t pre-push \
+        -t pre-rebase \
+        -t prepare-commit-msg && \
```
Comment (amolenaar) [inline base/Dockerfile:76]: It's a pitty there's no way to install pre-commit hooks in a more straightforward way.

**Review by amolenaar — APPROVED** [review]

File: docs/docs/git-hooks/git-hooks.md (hunk 2 above)
Comment (MoritzWeber0) [reply docs/docs/git-hooks/git-hooks.md:42]: > When in a Capella session, you do not have access to a command line, right? Then there's no way to update. updates could be done through a Jupyter session.

Might be worth to mention it, good point. I was thinking of any other device or Web IDEs (like the Gitlab one).

Comment (MoritzWeber0) [reply docs/docs/git-hooks/git-hooks.md:42]: It's possible to open text files in the Eclipse editor. Alternatively, it's also possible to use Jupyter, Web IDEs (like the Gitlab editor) or clone the repository on any other device. I wouldn't mention it here.

File: base/Dockerfile (hunk 3 above)
Comment (MoritzWeber0) [reply base/Dockerfile]: Had to reimplement due to a deadlock with `pre-commit`. It looks a bit better now.

**Pull request merged**